guppylang/checker/linearity_checker.py

```diff
@@ -419,6 +421,13 @@ def visit_BarrierExpr(self, node: BarrierExpr) -> None:
         self._visit_call_args(node.func_ty, node)
         self._reassign_inout_args(node.func_ty, node)
 
+    def visit_ResultExpr(self, node: ResultExpr) -> None:
```

Comment:
this is now necessary because arrays are affine and so we need the linearity checking to get involved correct?

Reply:
should we have some error test that such linearity errors are being caught?

Reply:
> this is now necessary because arrays are affine and so we need the linearity checking to get involved correct?

Exactly!

> should we have some error test that such linearity errors are being caught?

Yes, I'll add a "result after move" test 👍